plugins/win-capture/game-capture.c

```diff
@@ -592,21 +601,104 @@ static void load_placeholder_image(struct game_capture *gc)
 {
 	unload_placeholder_image(gc);
 
-	if (!dstr_is_empty(&gc->placeholder_img)) {
-		gs_image_file2_init(&gc->if2, gc->placeholder_img.array);
+	if (!dstr_is_empty(&gc->placeholder_image_path)) {
+		gs_image_file2_init(&gc->placeholder_image, gc->placeholder_image_path.array);
 
 		obs_enter_graphics();
-		gs_image_file2_init_texture(&gc->if2);
+		gs_image_file2_init_texture(&gc->placeholder_image);
 		obs_leave_graphics();
 	}
+	
+	TCHAR* translated_string = NULL;
+	size_t len = os_utf8_to_wcs(gc->placeholder_text.array, 0, NULL, 0);
+	if (len)
+	{
+		translated_string = malloc( (len+1)*2);
+		os_utf8_to_wcs(gc->placeholder_text.array, 0, &translated_string[0], len + 1);
+	} else {
+		return;
+	}
+
+	struct obs_video_info ovi;	
+	obs_get_video_info(&ovi);
+
+	gc->placeholder_text_height = ovi.base_height/8;
+	gc->placeholder_text_width = ovi.base_width;
+
+	BITMAPINFOHEADER bmphdr = { 0 };
+	bmphdr.biSize = sizeof(BITMAPINFOHEADER);
+	bmphdr.biWidth = gc->placeholder_text_width;
+	bmphdr.biHeight = -gc->placeholder_text_height;
+	bmphdr.biPlanes = 1;
+	bmphdr.biBitCount = 32;
+	bmphdr.biSizeImage = gc->placeholder_text_height * gc->placeholder_text_width * 4;
+	
+	HDC wdc = GetDC(NULL);
+	if (wdc) {
+		HDC memDC = CreateCompatibleDC ( wdc );
+
+		uint8_t * mdata = bmalloc(gc->placeholder_text_height*gc->placeholder_text_width*4);	
+		HBITMAP text_bitmap = CreateDIBSection(NULL, (PBITMAPINFO)&bmphdr, DIB_RGB_COLORS, &mdata, NULL, 0);
+		if (text_bitmap) {
+			SelectObject ( memDC, text_bitmap );
+			SetTextColor(memDC, 0x00FFFFFF);
+			SetBkColor(memDC, 0x00000000);
+
+			int string_width = 0;
+
+			HFONT   font;
+			LOGFONT LogFont = {0};
+			LogFont.lfStrikeOut = 0;
+			LogFont.lfUnderline = 0;
+			LogFont.lfEscapement = 0;
+			LogFont.lfQuality = 0x05;
+			LogFont.lfItalic = 0;
+			LogFont.lfHeight = gc->placeholder_text_height;
+
+			while (true) {
+				font = CreateFontIndirect(&LogFont);
+				SelectObject(memDC, font);
+				
+				GetTextExtentPoint32(memDC, translated_string, len, &string_width);
+				if (string_width < gc->placeholder_text_width) 
+					break;
+				
+				DeleteObject(font);
+				LogFont.lfHeight = LogFont.lfHeight - 5;
```

Comment:
why are we substracting 5? Not clear, please use a meaning full constant variable.